conda_smithy/linter/lints.py

```diff
@@ -983,3 +987,100 @@ def sort_osx(versions):
         ):
             if sdk_lint not in lints:
                 lints.append(sdk_lint)
+
+
+def lint_recipe_is_parsable(
+    recipe_text: str,
+    lints: List[str],
+    hints: List[str],
+    recipe_version: int = 0,
+):
+    parse_vars = []
+    parser_names = []
+
+    if recipe_version == 0:
+        try:
+            from conda_forge_tick.recipe_parser import CondaMetaYAML
+        except ImportError:
+            parses_with_conda_forge_tick = None
+            pass
+        else:
+            try:
+                CondaMetaYAML(recipe_text)
+            except Exception as e:
+                logger.warning(
+                    "Error parsing recipe with conda-forge-tick (the bot): %s",
+                    repr(e),
+                    exc_info=e,
+                )
+                parses_with_conda_forge_tick = False
+            else:
+                parses_with_conda_forge_tick = True
+
+        parse_vars.append(parses_with_conda_forge_tick)
+        parser_names.append("conda-forge-tick (the bot)")
+
+        try:
+            from souschef.recipe import Recipe
+        except ImportError:
+            parses_with_souschef = None
+            pass
+        else:
+            with tempfile.TemporaryDirectory() as tmpdir:
+                recipe_file = os.path.join(tmpdir, "meta.yaml")
+                with open(recipe_file, "w") as f:
+                    f.write(recipe_text)
```

Comment:
Souschef modifies the recipe file on parsing? 🤔

Reply:
Likely not but why take the risk?